stdlib/Test/test/runtests.jl

```diff
@@ -1892,3 +1892,174 @@ end
+                @test occursin("Evaluated: !(isempty(Any[]) && 1 == 1)", str)
+                @test !occursin("&& ...", str)  # Should not short-circuit when negated
+            end
+
+            # Test negated OR expression
+            let str = sprint(show, negated_fails[2])
+                @test occursin("Expression: !(true || false)", str)
+                # Note: This might not show "Evaluated:" because it's all literals
+            end
+
+            # Test negated expression with literals (should not show "Evaluated:")
+            let str = sprint(show, negated_fails[3])
+                @test occursin("Expression: !(1 == 1 && 2 == 2)", str)
+                @test !occursin("Evaluated:", str)  # Literals shouldn't show evaluation
+            end
+
+            # Test negated expression with variables (should show "Evaluated:" with !)
+            let str = sprint(show, negated_fails[4])
+                @test occursin("Expression: !(x < y && y > x)", str)
+                @test occursin("Evaluated: !(1 < 2 && 2 > 1)", str)
+            end
+        end
+
+        # Test that passing negated expressions don't show expansion details
+        @test !(false && true)  # !(false) = true - should pass
+        @test !((1 == 2) || (3 == 4))  # !(false || false) = !(false) = true - should pass
+    end
+    @testset "Nested negation expansion" begin
+        # Test simple nested negation cases
+        let nested_fails = @testset NoThrowTestSet begin
+                # Cases that should fail (evaluate to false)
+                @test !(!false)  # !(!false) = !(true) = false - should fail
+                @test !(!(true && false))  # !(!(true && false)) = !(!(false)) = !true = false - should fail
+                x, y = 1, 2
+                @test !((x < y) && !(x > y))
+                x = [1]
+                @test !(true && !(isempty(x) && isempty(x)))
+            end
+
+            for nf in nested_fails
+                @test nf isa Test.Fail
+            end
+
+            # Test display of nested negation shows proper expansion
+            let str = sprint(show, nested_fails[1])
+                @test occursin("Expression: !(!false)", str)
+                # For simple literals, may not show "Evaluated" line
+            end
+
+            let str = sprint(show, nested_fails[2])
+                @test occursin("Expression: !(!(true && false))", str)
+                # Should show some expansion since it involves logical operators
+            end
+
+            let str = sprint(show, nested_fails[3])
+                @test occursin("Evaluated: !(1 < 2 && !(1 > 2))", str)
+            end
+
+            let str = sprint(show, nested_fails[4])
+                @test occursin("Evaluated: !(true && !(isempty([1]) && ...))", str)
+            end
+
+            @test true && !(false && false)
+            @test true && !(false && true)
+            @test true && !(false && !false)
+            @test true && !(false && !true)
```

Comment:
What are these for?

Reply:
Testing that the new code didn't break various logical structures